src/App/Osu/OsuDatabaseBeatmap.cpp

```diff
@@ -358,7 +366,7 @@ OsuDatabaseBeatmap::PRIMITIVE_CONTAINER OsuDatabaseBeatmap::loadPrimitiveObjects
 							colorOffset += (type >> 4) & 7; // special case 3: "Bits 4-6 (16, 32, 64) form a 3-bit number (0-7) that chooses how many combo colours to skip."
 						}
 
-						if (type & 0x1) // circle
+						if ((type & 0x1) || (osu_mod_no_sliders.getBool() && (type & 0x2))) // circle
```

Comment:
I know I suggested putting this logic inside loadPrimitiveObjects(), but actually this is a very bad idea since it will infect all pp calcs everywhere (meaning if a user recalculated their profile pp while this mod is enabled, all scores would suddenly be calculated as if having had that mod enabled).

Note: Experimental mods are _never_ considered for profile pp recalcs, that's why they are called "Experimental". However, having an experimental mod enabled should not affect unrelated calculations in other parts of the code as drastically as this one.

You want to have correct pp values if sliders are turned into circles, but it shouldn't affect _everything_ globally. This is problematic because there is no infrastructure to support that logic currently. As mentioned, I don't care for pp/star accuracy when experimental mods are used, which is why this is the way it is. Therefore this can't be merged as is.

Reply:
My suggestion would be to ignore accurate pp/stars and put the if-condition inside loadGameplay() instead.